cmd/guacone/cmd/s3.go

```diff
@@ -0,0 +1,154 @@
+			MessageProviderHost: s3Opts.mpHostname,
+			MessageProviderPort: s3Opts.mpPort,
+			Queues:              s3Opts.queues,
+			Region:              s3Opts.region,
+		})
+		if err != nil {
+			logger.Errorf("unable to create s3 collector: %v", err)
+			os.Exit(1)
+		}
+
+		err = collector.RegisterDocumentCollector(s3Collector, s3.S3CollectorType)
+		if err != nil {
+			logger.Errorf("unable to register s3 collector: %v\n", err)
+			os.Exit(1)
+		}
+
+		csubClient, err := csub_client.NewClient(viper.GetString("csub-addr"))
+		if err != nil {
+			logger.Infof("collectsub client initialization failed, this ingestion will not pull in any additional data through the collectsub service: %v", err)
+			csubClient = nil
+		} else {
+			defer csubClient.Close()
+		}
+
+		errFound := false
+
+		_, s3ctx := errgroup.WithContext(ctx)
+		emit := func(d *processor.Document) error {
+			err := ingestor.Ingest(s3ctx, d, viper.GetString("gql-addr"), csubClient)
+
+			if err != nil {
+				errFound = true
+				return fmt.Errorf("unable to ingest document: %w", err)
+			}
+			return nil
+		}
+
+		errHandler := func(err error) bool {
+			if err == nil {
+				logger.Info("collector ended gracefully")
+				return true
+			}
+			logger.Errorf("collector ended with error: %v", err)
+			return false
+		}
+
+		if err := collector.Collect(ctx, emit, errHandler); err != nil {
+			logger.Fatal(err)
+		}
+
+		if errFound {
+			logger.Fatalf("completed ingestion with error")
+		} else {
+			logger.Infof("completed ingestion")
+		}
+	},
+}
+
+func validateS3Opts(s3hostname string, s3port string, mp string, mpHostname string, mpPort string, queues string, region string) (s3Options, error) {
+	var opts s3Options
+	if len(s3hostname) == 0 {
+		return opts, fmt.Errorf("expected s3 hostname")
+	}
+	if len(s3port) == 0 {
+		return opts, fmt.Errorf("expected s3 port")
+	}
+	if len(mpHostname) == 0 {
+		return opts, fmt.Errorf("expected hostname for message provider")
+	}
+	if len(mpPort) == 0 {
+		return opts, fmt.Errorf("expected port for message provider")
+	}
+	if len(queues) == 0 {
+		return opts, fmt.Errorf("expected at least one queue")
+	}
+
+	opts = s3Options{s3hostname, s3port, region, queues, mp, mpHostname, mpPort}
+
+	return opts, nil
+}
+
+func init() {
+	s3Cmd.Flags().StringVarP(&s3Opts.s3hostname, "s3-host", "", "", "hostname for s3")
+	s3Cmd.Flags().StringVarP(&s3Opts.s3port, "s3-port", "", "", "port for s3")
+	s3Cmd.Flags().StringVarP(&s3Opts.mp, "mp", "m", "", "message provider (sqs or kafka, defaults to kafka)")
+	s3Cmd.Flags().StringVarP(&s3Opts.mpHostname, "mp-host", "", "", "hostname for the message provider (sqs/kafka)")
+	s3Cmd.Flags().StringVarP(&s3Opts.mpPort, "mp-port", "", "", "port for the message provider (sqs/kafka)")
+	s3Cmd.Flags().StringVarP(&s3Opts.queues, "queues", "", "", "comma-separated list of queue/topic names")
+	s3Cmd.Flags().StringVarP(&s3Opts.region, "region", "", "us-east-1", "aws region, defaults to us-east-1")
```

Comment:
Please add these to `pkg/cli/store.go` and access them the same way as the other commands. Also add "s3-" to the front of all of them.